crates/hir-ty/src/diagnostics/match_check/pat_analysis.rs

```diff
@@ -328,7 +325,7 @@ impl<'db> PatCx for MatchCheckCtx<'db> {
         self.exhaustive_patterns
     }
     fn is_min_exhaustive_patterns_feature_on(&self) -> bool {
-        self.min_exhaustive_patterns
+        true
```

Comment:
Should I bump up `ra-ap-rustc_pattern_analysis` to latest instead?

Reply:
There is a PR upstream enabling the in-tree pattern analysis in the rust repo (to prevent us from going out of sync), so these two PRs will likely collide. Might be best if either PR waits on the other (and if we pick this one here first then yes we should bump the dependency here), cc @Nadrieril

Reply:
Since the rust-lang/rust/#128922 is in homu queue, maybe I should wait for it?